src/providers/network-connection/network-connection-provider.test.tsx

```diff
@@ -0,0 +1,199 @@
+        SetupSutResults,
+        "networkConnectionResults"
+    > = {
+        all: [] as NetworkConnection[],
+    };
+
+    function TestComponent() {
+        const connection = useContext(NetworkConnectionContext);
+        networkConnectionResults.all.push(connection);
+        networkConnectionResults.current = connection;
+
+        return <div></div>;
+    }
+
+    return {
+        TestComponent,
+        networkConnectionResults,
+    };
+};
+
+// #endregion Setup
+
+describe("NetworkConnectionProvider", () => {
+    it("renders initial network connection state", () => {
+        // Arrange
+        const connection: NetworkConnection = {
+            isOnline: true,
+        };
+        const { networkConnectionResults, TestComponent } = setupSut({
+            mockConnections: [connection],
+        });
+
+        // Act
+        render(
+            <NetworkConnectionProvider>
+                <TestComponent />
+            </NetworkConnectionProvider>
+        );
+
+        // Assert
+        expect(networkConnectionResults.all.length).toEqual(2);
+        expect(networkConnectionResults.current).toEqual(connection);
+    });
+
+    it("adds an event listener", () => {
+        // Arrange
+        const addEventListener = jest.fn();
+        const { TestComponent } = setupSut({
+            mockConnections: [{ addEventListener }],
+        });
+
+        // Act
+        render(
+            <NetworkConnectionProvider>
+                <TestComponent />
+            </NetworkConnectionProvider>
+        );
+
+        // Assert
+        expect(addEventListener).toBeCalled();
+    });
+
+    describe("when change event is called", () => {
+        it("loads network connection into state", () => {
+            // Arrange
+            let loadNetworkInformationCallback = () => {};
+
+            const mockConnections: Array<NetworkConnection> = [
+                {
+                    isOnline: true,
+                },
+                {
+                    isOnline: true,
+                    addEventListener: (
+                        event: "change",
+                        callback: VoidFunction
+                    ) => {
+                        loadNetworkInformationCallback = callback;
+                    },
+                },
+                {
+                    isOnline: false,
+                },
+            ];
+
+            const { networkConnectionResults, TestComponent } = setupSut({
+                mockConnections,
+            });
+
+            // Act
+            render(
+                <NetworkConnectionProvider>
+                    <TestComponent />
+                </NetworkConnectionProvider>
+            );
+
+            act(() => loadNetworkInformationCallback());
+
+            // Assert
+            expect(networkConnectionResults.all.length).toEqual(
+                mockConnections.length
+            );
+            expect(networkConnectionResults.current).toEqual(
```

Comment:
This test is pretty abstracted - what is this actually asserting on? Since the `loadNetworkInformationCallback` function was called, it updates the current NetworkConnection state value to be the last value? (`{ isOnline: false }`?)